### PR TITLE
Rm react-test-renderer

### LINKED ISSUE #774: Remove use of react-test-renderer
We currently have 2 test renderers in Psammead and Simorgh - `react-test-renderer` and `@testing-library/react`.

Making use of 2 test renderers will lead to confusion about which one to use and when, especially when they have different ideas about how to write tests.

## Solution
Remove `react-test-renderer` and all use of it. Refactor shallow render tests because `@testing-library/react` does not have or encourage shallow rendering.

---

Some of the benefits of using `@testing-library/react` include:
 - encourages better testing practices and writing more maintainable tests 
 - is designed to enable and encourage writing tests that use components as the end users use them
 - it supports querying for elements in ways that encourage accessibility in components

Find out more about [@testing-library/react](https://testing-library.com/docs/react-testing-library/intro).

This article briefly describes and compares React test renders https://codeburst.io/revisiting-react-testing-in-2019-ee72bb5346f4

Here is more on why [`@testing-library/react` doesn't do shallow rendering](https://kentcdodds.com/blog/why-i-never-use-shallow-rendering).

### PR DESCRIPTION
**Pull request by mykeels**
Resolves https://github.com/bbc/psammead/issues/774

Duplicate of #4377

**Overall change:** _Remove use of `react-test-renderer`, and use `@testing-library/react` instead._

**Code changes:**

- _Remove use of `react-test-renderer`, and use `@testing-library/react` instead._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

**Comment by mykeels** [comment]
Please do NOT "Update Branch" via GitHub's UI

**Review by EinsteinNjoroge — APPROVED** [review]

File: src/app/containers/ATIAnalytics/canonical/index.test.jsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { create, act } from 'react-test-renderer';
+import { render as create, act } from '@testing-library/react';
 import { render } from 'enzyme';
```
Comment (jroebu14) [inline src/app/containers/ATIAnalytics/canonical/index.test.jsx:3]: is there an issue to look at removing enzyme too?

File: src/app/containers/ChartbeatAnalytics/__snapshots__/index.test.jsx.snap [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (jroebu14) [inline src/app/containers/ChartbeatAnalytics/__snapshots__/index.test.jsx.snap]: I don't think these should be snapshots

File: src/app/containers/ChartbeatAnalytics/index.test.jsx [hunk 3]
```diff
@@ -120,35 +118,31 @@ describe('Charbeats Analytics Container', () => {
...
-      .toJSON();
-
-    expect(tree).toMatchSnapshot();
+    const { container } = render(
+      <ContextWrap platform="canonical" pageType="article" origin="bbc.com">
+        <ChartbeatAnalytics data={frontPageData} />
+      </ContextWrap>,
+    );
+
+    expect(container.firstChild).toMatchSnapshot();
```
Comment (jroebu14) [inline src/app/containers/ChartbeatAnalytics/index.test.jsx]: based on the title of this test I would change the assertion to `expect(container.firstChild).toBeNull();`

File: src/app/containers/ChartbeatAnalytics/index.test.jsx [hunk 4]
```diff
@@ -120,35 +118,31 @@ describe('Charbeats Analytics Container', () => {
...
+        platform="canonical"
+        pageType="article"
+        origin={localBaseUrl}
+      >
+        <ChartbeatAnalytics data={frontPageData} />
+      </ContextWrap>,
+    );
 
-    expect(tree).toMatchSnapshot();
+    expect(container.firstChild).toMatchSnapshot();
```
Comment (jroebu14) [inline src/app/containers/ChartbeatAnalytics/index.test.jsx]: based on the title of this test I would change the assertion to `expect(container.firstChild).toBeNull();`

**Review by rhenshaw56 — APPROVED** [review]

File: src/app/containers/ATIAnalytics/canonical/index.test.jsx (hunk 1 above)
Comment (mykeels) [reply src/app/containers/ATIAnalytics/canonical/index.test.jsx:3]: Now tracked in https://github.com/bbc/simorgh/issues/4491

File: src/app/containers/ChartbeatAnalytics/__snapshots__/index.test.jsx.snap (hunk 2 above)
Comment (mykeels) [reply src/app/containers/ChartbeatAnalytics/__snapshots__/index.test.jsx.snap]: Fixed!

File: src/app/containers/ChartbeatAnalytics/index.test.jsx (hunk 3 above)
Comment (mykeels) [reply src/app/containers/ChartbeatAnalytics/index.test.jsx]: Fixed!

File: src/app/containers/ChartbeatAnalytics/index.test.jsx (hunk 4 above)
Comment (mykeels) [reply src/app/containers/ChartbeatAnalytics/index.test.jsx]: Fixed!

File: src/app/containers/ChartbeatAnalytics/__snapshots__/index.test.jsx.snap [hunk 5]
```diff
@@ -1,7 +1,3 @@
...
 
-exports[`Charbeats Analytics Container should call AmpCharbeatsBeacon when platform is amp and toggle enabled for chartbeat on live 1`] = `"amp-return-value"`;
-
-exports[`Charbeats Analytics Container should return null when toggle is disbaled for live 1`] = `null`;
-
-exports[`Charbeats Analytics Container should return null when toggle is disbaled for localhost 1`] = `null`;
+exports[`Charbeats Analytics Container should call AmpCharbeatsBeacon when platform is amp and toggle enabled for chartbeat on live 1`] = `amp-return-value`;
```
Comment (jroebu14) [inline src/app/containers/ChartbeatAnalytics/__snapshots__/index.test.jsx.snap]: can we get rid of this snapshot too please?

**Review by jroebu14 — CHANGES_REQUESTED** [review]
Nice work. Just one more change and this is good!

**Review by jroebu14 — APPROVED** [review]
Looks good!

**Pull request merged**